File: nodejs/ipsd.js
// Author: d1y<chenhonzhou@gmail.com>

const pid = "2"

const calg = "12345678"

const createMD5 = data=> {
  var crypto = require('crypto')
  return crypto.createHash('md5').update(data).digest("hex")
}

const createPassword = p=> {
  const r = pid + p + calg
  const output = createMD5(r) + calg + pid
  return output
}

;(()=> {
  const args = process.argv
  if (args.length == 2) {
    console.log("请传入密码")
    return
  }
  const p = args[2]
  const output = createPassword(p)
  console.log(output)
})()

File: python/ipsd.py
# Author: d1y<chenhonzhou@gmail.com>

import sys, hashlib

pid = "2"

calg = "12345678"

def createMD5(str):
  hl = hashlib.md5()
  hl.update(str.encode(encoding='utf-8'))
  hex = hl.hexdigest()
  return hex

def createPassword(str):
  r = pid + str + calg
  output = createMD5(r) + calg + pid
  return output

if __name__ == "__main__":
  args = sys.argv
  if len(args) == 1:
    print("请传入密码")
  else:
    p = args[1]
    output = createPassword(p)
    print(output)